src/MessagePack.Generator/MessagepackCompiler.cs

```diff
@@ -12,30 +21,97 @@ public class MessagepackCompiler : ConsoleAppBase
     {
         private static async Task Main(string[] args)
         {
+            var instance = MSBuildLocator.RegisterDefaults();
+            AssemblyLoadContext.Default.Resolving += (assemblyLoadContext, assemblyName) =>
+            {
+                var path = Path.Combine(instance.MSBuildPath, assemblyName.Name + ".dll");
+                if (File.Exists(path))
+                {
+                    return assemblyLoadContext.LoadFromAssemblyPath(path);
+                }
+
+                return null;
+            };
+
             await Host.CreateDefaultBuilder()
                 .ConfigureLogging(logging => logging.ReplaceToSimpleConsole())
-                .RunConsoleAppFrameworkAsync<MessagepackCompiler>(args)
-                .ConfigureAwait(false);
+                .RunConsoleAppFrameworkAsync<MessagepackCompiler>(args);
         }
 
         public async Task RunAsync(
-            [Option("i", "Input path of analyze csproj or directory, if input multiple csproj split with ','.")]string input,
-            [Option("o", "Output file path(.cs) or directory(multiple generate file).")]string output,
-            [Option("c", "Conditional compiler symbols, split with ','.")]string conditionalSymbol = null,
-            [Option("r", "Set resolver name.")]string resolverName = "GeneratedResolver",
-            [Option("n", "Set namespace root name.")]string @namespace = "MessagePack",
-            [Option("m", "Force use map mode serialization.")]bool useMapMode = false,
-            [Option("ms", "Generate #if-- files by symbols, split with ','.")]string multipleIfDirectiveOutputSymbols = null)
+            [Option("i", "Input path of analyze MSBuild project file or directory, if input multiple project files split with ','.")] string input,
+            [Option("o", "Output file path(.cs) or directory(multiple generate file).")] string output,
+            [Option("c", "Conditional compiler symbols, split with ','.")] string? conditionalSymbol = null,
+            [Option("r", "Set resolver name.")] string resolverName = "GeneratedResolver",
+            [Option("n", "Set namespace root name.")] string @namespace = "MessagePack",
+            [Option("m", "Force use map mode serialization.")] bool useMapMode = false,
+            [Option("ms", "Generate #if-- files by symbols, split with ','.")] string? multipleIfDirectiveOutputSymbols = null)
+        {
+            // Add this to your C# console app's Main method to give yourself
+            // a CancellationToken that is canceled when the user hits Ctrl+C.
+            var cts = new CancellationTokenSource();
```

Comment:
We can use `this.Context.CancellationToken` that provided by ConsoleAppFramework(and GenericHost's Console lifetime)